internal/worker/apiserver/observers.go

```diff
@@ -4,53 +4,87 @@
 package apiserver
 
 import (
+	"context"
+
 	"github.com/juju/clock"
 	"github.com/juju/errors"
-	"github.com/juju/pubsub/v2"
 
 	"github.com/juju/juju/agent"
 	"github.com/juju/juju/apiserver"
 	"github.com/juju/juju/apiserver/observer"
 	"github.com/juju/juju/apiserver/observer/metricobserver"
+	"github.com/juju/juju/core/model"
 	internallogger "github.com/juju/juju/internal/logger"
+	"github.com/juju/juju/internal/services"
 )
 
 func newObserverFn(
 	agentConfig agent.Config,
+	domainServicesGetter services.DomainServicesGetter,
 	clock clock.Clock,
-	hub *pubsub.StructuredHub,
 	metricsCollector *apiserver.Collector,
 ) (observer.ObserverFactory, error) {
-	var observerFactories []observer.ObserverFactory
-
-	// Common logging of RPC requests
-	observerFactories = append(observerFactories, func() observer.Observer {
-		logger := internallogger.GetLogger("juju.apiserver")
-		ctx := observer.RequestObserverContext{
-			Clock:  clock,
-			Logger: logger,
-			Hub:    hub,
-		}
-		return observer.NewRequestObserver(ctx)
-	})
-
 	// Metrics observer.
 	metricObserver, err := metricobserver.NewObserverFactory(metricobserver.Config{
 		Clock:            clock,
-		MetricsCollector: metricCollectorWrapper{collector: metricsCollector},
+		MetricsCollector: metricCollector{collector: metricsCollector},
 	})
 	if err != nil {
 		return nil, errors.Annotate(err, "creating metric observer factory")
 	}
-	observerFactories = append(observerFactories, metricObserver)
 
-	return observer.ObserverFactoryMultiplexer(observerFactories...), nil
+	return observer.ObserverFactoryMultiplexer(
+		requestLogger(clock),
+		metricObserver,
+		agentPresence(domainServicesGetter),
+	), nil
+}
```

Comment:
Ah, I see now why we have a stub-like `requestLogger` observer.

If `Multiplexer` means what I think it means, this probably isn't the best pattern for what we're after. A middleware pattern is probably more suitable